crates/starknet-devnet/tests/test_memory_test.rs

```diff
@@ -0,0 +1,47 @@
+pub mod common;
+
+mod memory_test {
+
+    use peak_alloc::PeakAlloc;
+
+    use crate::common::background_devnet::BackgroundDevnet;
+
+    #[global_allocator]
+    static PEAK_ALLOC: PeakAlloc = PeakAlloc;
+
+    static DUMMY_ADDRESS: u128 = 1;
+    static DUMMY_AMOUNT: u128 = 1;
+
+    async fn mint_iter(f: &str) -> f32 {
+        // reset peak usage
+        PEAK_ALLOC.reset_peak_usage();
+
+        let devnet = BackgroundDevnet::spawn_with_additional_args(&["--state-archive-capacity", f])
+            .await
+            .expect("Could not start Devnet");
+
+        for n in 1..=5_000 {
+            println!("n: {:?}", n);
+            let mint_tx_hash = devnet.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
+            println!("mint_tx_hash: {:?}", mint_tx_hash);
+        }
+
+        // set and return peak usage after recurring mint run
+        PEAK_ALLOC.peak_usage_as_mb()
+    }
+
+    #[tokio::test]
+    async fn test_memory_limit() {
```

Comment:
The test name is not informative enough. My suggestion is to call it something like `test_full_state_archive_capacity_presents_no_increase`